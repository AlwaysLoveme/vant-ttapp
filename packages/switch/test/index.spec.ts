import path from 'path';
import simulate from 'miniprogram-simulate';

describe('switch', () => {
  const VanSwitch = simulate.load(
    path.resolve(__dirname, '../../switch/index'),
    'van-switch',
    {
      rootPath: path.resolve(__dirname, '../../'),
    }
  );

  test('should allow to custom active-value and inactive-value', async () => {
    const comp = simulate.render(
      simulate.load({
        usingComponents: {
          'van-switch': VanSwitch,
        },
        template: `
          <van-switch
            id="wrapper"
            checked="{{ checked }}"
            active-color="#07c160"
            inactive-color="#ee0a24"
            active-value="on"
            inactive-value="off"
            bind:change="onChange"
          />
        `,
        data: {
          checked: 'on',
        },
        methods: {
          onChange({ detail }) {
            this.setData({ checked: detail });
          },
        },
      })
    );
    comp.attach(document.createElement('parent-wrapper'));

    const wrapper = comp.querySelector('#wrapper');
    const btn = wrapper?.querySelector('.van-switch');
    expect((btn as any).toJSON()).toMatchSnapshot();
    btn?.dispatchEvent('tap');
    await simulate.sleep(10);
    expect((btn as any).toJSON()).toMatchSnapshot();
    expect(comp.data.checked).toEqual('off');
  });
});
