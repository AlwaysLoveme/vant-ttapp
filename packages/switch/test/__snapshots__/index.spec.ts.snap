// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switch should allow to custom active-value and inactive-value 1`] = `
<wx-view
  class="van-switch van-switch--on custom-class"
  style="font-size:30px;background-color:#07c160"
  bind:tap="onClick"
>
  <wx-view
    class="van-switch__node node-class"
  />
</wx-view>
`;

exports[`switch should allow to custom active-value and inactive-value 2`] = `
<wx-view
  class="van-switch custom-class"
  style="font-size:30px;background-color:#ee0a24"
  bind:tap="onClick"
>
  <wx-view
    class="van-switch__node node-class"
  />
</wx-view>
`;
